client/rest-client/src/main/java/org/infinispan/client/rest/impl/okhttp/RestClientOkHttp.java

```diff
@@ -64,13 +66,31 @@ public RestClientOkHttp(RestClientConfiguration configuration) {
       builder
             .connectTimeout(configuration.connectionTimeout(), TimeUnit.MILLISECONDS)
             .readTimeout(configuration.socketTimeout(), TimeUnit.MILLISECONDS).followRedirects(configuration.followRedirects());
-
-      SSLContext sslContext = configuration.security().ssl().sslContext();
-      if (sslContext != null) {
-         builder.sslSocketFactory(sslContext.getSocketFactory(), (X509TrustManager) configuration.security().ssl().trustManagers()[0]);
-         HostnameVerifier hostnameVerifier = configuration.security().ssl().hostnameVerifier();
-         if (hostnameVerifier != null) {
-            builder.hostnameVerifier(hostnameVerifier);
+      SslConfiguration ssl = configuration.security().ssl();
+      if (ssl.enabled()) {
+         SSLContext sslContext = ssl.sslContext();
+         if (sslContext == null) {
+            SslContextFactory sslContextFactory = new SslContextFactory()
+                  .keyStoreFileName(ssl.keyStoreFileName())
+                  .keyStorePassword(ssl.keyStorePassword())
+                  .keyStoreType(ssl.keyStoreType())
+                  .trustStoreFileName(ssl.trustStoreFileName())
+                  .trustStorePassword(ssl.trustStorePassword())
+                  .trustStoreType(ssl.trustStoreType())
+                  .classLoader(Thread.currentThread().getContextClassLoader())
+                  .useNativeIfAvailable(false);
```

Comment:
No openssl for the RestClient?

Reply:
Yes, but we configure the SSLCotnext it ourselves instead of doing it OkHttpClient's way.